src/Serialization.cpp

```diff
@@ -0,0 +1,1310 @@
+#include "Serialization.h"
+#include "Schedule.h"
+#include "halide_ir_generated.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "FindCalls.h"
+#include "Func.h"
+#include "Function.h"
+#include "IR.h"
+#include "RealizationOrder.h"
+#include <fstream>
+#include <iostream>
+#include <map>
+
+namespace Halide {
+namespace Internal {
+class Serializer {
+public:
+    Serializer() = default;
+
+    void serialize(const Pipeline &pipeline, const std::string &filename);
+
+    const std::unordered_map<std::string, Parameter> &get_external_parameters() const {
```

Comment:
(Pretty much all the comments I made for Deserialization.cpp apply here too, so I won't repeat them)